monix-logback-core/src/main/scala/com/softwaremill/correlator/CorrelationId.scala

```diff
@@ -4,25 +4,19 @@ import java.{util => ju}
 
 import cats.data.{Kleisli, OptionT}
 import ch.qos.logback.classic.util.LogbackMDCAdapter
+import com.softwaremill.correlator.CorrelationId._
 import monix.eval.Task
 import monix.execution.misc.Local
-import org.http4s.util.CaseInsensitiveString
-import org.http4s.{HttpRoutes, Request}
 import org.slf4j.{Logger, LoggerFactory, MDC}
 
 import scala.util.Random
 
 /**
-  * Correlation id support. The `init()` method should be called when the application starts.
-  * See [[https://blog.softwaremill.com/correlation-ids-in-scala-using-monix-3aa11783db81]] for details.
-  */
-class CorrelationId(
-    val headerName: String = "X-Correlation-ID",
-    logStartRequest: (String, Request[Task]) => Task[Unit] = (cid, req) =>
```

Comment:
didn't we loose the functionality of logging a starting request?

Reply:
Yes we did, I wrote about it ;)
> There is also a difference from the business perspective as we now cannot set logStartRequest. I think that it can be easily added combining this middleware with another one and IMO it violated SRP.

Reply:
well ... if it can be easily added, maybe we should add it?

SRP is great but then we loose part of the functionality, which takes precendece over SRP ;)